libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -404,6 +416,50 @@ static bool _createKeepAliveOperation( const IotMqttNetworkInfo_t * pNetworkInfo
 
 /*-----------------------------------------------------------*/
 
+static IotNetworkError_t _createNetworkConnection( const IotMqttNetworkInfo_t * pNetworkInfo,
+                                                   void ** pNetworkConnection )
+{
+    IOT_FUNCTION_ENTRY( IotNetworkError_t, IOT_NETWORK_SUCCESS );
+
+    /* Network info must not be NULL. */
+    if( pNetworkInfo == NULL )
+    {
+        IotLogError( "Network information cannot be NULL." );
+
+        IOT_SET_AND_GOTO_CLEANUP( IOT_NETWORK_BAD_PARAMETER );
+    }
+    else
+    {
+        EMPTY_ELSE_MARKER;
+    }
+
+    /* Create a new network connection if requested. Otherwise, copy the existing
+     * network connection. */
+    if( pNetworkInfo->createNetworkConnection == true )
+    {
+        status = pNetworkInfo->pNetworkInterface->create( pNetworkInfo->u.setup.pNetworkServerInfo,
+                                                          pNetworkInfo->u.setup.pNetworkCredentialInfo,
+                                                          pNetworkConnection );
+
+        if( status != IOT_NETWORK_SUCCESS )
+        {
+            IotLogError( "Failed to create network connection: %d", status );
+
+            IOT_GOTO_CLEANUP();
+        }
```

Comment:
Nit: EMPTY_ELSE_MARKER after the if block

Reply:
Eliminated after reverting to earlier code which sets `pCreatedNetworkConnection`.